docs/DeveloperGuide.md

```diff
@@ -8,8 +8,189 @@
+
+![Timetable Class Diagram](./images/Timetable_Class.png)
+
+In chronological order, the following diagrams illustrate the flow of the program for adding lessons, deleting lessons and displaying timetables to the user.
+
+![Timetable Add Lesson Sequence Diagram](./images/Timetable_addLesson_Sequence.png)
+![Timetable Delete Lesson Sequence Diagram](./images/Timetable_deleteLesson_Sequence.png)
+![Timetable Print Timetable Sequence Diagram](./images/Timetable_printTimetable_Sequence.png)
+
+### Delete History
+
+To help users recall the modules that they had recently deleted, the Delete History feature allows them to view up to 5 most
+recently deleted module mappings. This can help them to add the module mappings back to the lists without having to search for the specific
+module code again. 
+
+The UserDeletedModules class has an ArrayDeque, which stores the recently deleted module mappings.
+When the user deletes a module mapping, it will be added to the ArrayDeque.
+If the ArrayDeque already contains 5 module mappings, the last one (least recent) will be deleted, before the addition of a new module mapping.
+Do note that delete history information is not stored upon exiting the app (ie. it is not stored in User Storage).
+
+The following diagram illustrates the relationship between UserUniversityListManager and UserDeletedModules classes.
+
+![User Deleted Modules Class Diagram](./images/UserDeletedModules_Class.png)
+
+The following diagram illustrates the flow of the program, when a user deletes a module. 
+
+![User Deleted Modules Sequence Diagram](./images/UserDeletedModules_Sequence.png)
+
+### Ui
+
+The Ui class is the cornerstone of the Duke program to facilitate interaction with the user. It is used to scan and collect user input, print error messages to the user upon invalid input commands, 
+and display the appropriate acknowledgements or required information based on the user's command.
+
+The following diagram illustrates the methods within the Ui class that can be invoked by the other classes in Duke for the purpose of user interaction.
+
+![Ui Class Diagram](./images/Ui_Class.png)
+
+### Commands
+
+To interact with easySEP, users have to input commands specified with parameters to perform operations which will be passed to the CommandParser to generate a corresponding Command according to their input.
+The Command will then be executed to perform the operation on the other classes managing the timetables and databases. Error checking is handled to throw InvalidUserCommandException if the user's input does not match
+the specified Command format. To deal with parsing parameters, spaces in University names and Module codes are to be replaced with underscores.
+
+The following class diagram illustrates the relationship between Command class and its subclasses as well as other classes related to Commands.
+![Command Class Diagram](./images/Command_class.png)
```

Comment:
Your command class diagram is not rendered on the website view (https://ay2223s1-cs2113-w13-2.github.io/tp/DeveloperGuide.html), might want to look into this